src/Rules/Present.php

```diff
@@ -20,6 +20,20 @@ class Present extends Rule
      */
     public function check($value): bool
     {
+        $this->setAttributeAsRequired($this->attribute);
+
         return $this->validation->hasValue($this->attribute->getKey());
     }
+
+    /**
+     * Set attribute is required if $this->attribute is set
+     *
+     * @return void
+     */
+    protected function setAttributeAsRequired()
```

Comment:
This is a duplicate of `Required::setAttributeAsRequired`. If you want, I can extract the method in a trait, and reuse it in both classes